Integrations/integration-VxStream.yml

```diff
@@ -0,0 +1,416 @@
+                var newKey = mapper[key] || key;
+                res[newKey] = obj[key];
+            });
+            return res;
+        }
+    }
+
+    function createTableEntry(name, rawResponse, table, context, headers) {
+        return {
+            Type: entryTypes.note,
+            ContentsFormat: formats.json,
+            Contents: rawResponse,
+            ReadableContentsFormat: formats.markdown,
+            HumanReadable: tableToMarkdown(name, table, headers),
+            EntryContext: context
+        };
+    }
+
+    var sendRequest = function(uri) {
+        var requestUrl = serverUrl + uri
+
+        var res = http(
+            requestUrl,
+            {
+                Method: 'GET',
+                Headers: {
+                    'Authorization': [auth],
+                    'User-Agent': ['VxStream']
+                }
+            },
+            insecure,
+            proxy
+        );
+
+        if (res.StatusCode < 200 || res.StatusCode >= 300) {
+            throw 'Request Failed.\nStatus code: ' + res.StatusCode + '.\nBody: ' + JSON.stringify(res) + '.';
+        }
+
+        return JSON.parse(res.Body);
+    };
+
+    function scan(hash) {
+        var res = sendRequest('/api/scan/' + hash);
+        if (res.response_code !== 0) {
+            return entryError(res.response_code, res.response.error);
+        }
+
+        var response = res.response;
+
+        // create table from response
+        var tableMapper = {
+           threatlevel: 'threat level',
+           total_network_connections: 'total network connections',
+           targeturl: 'target url',
+           classification_tags: 'classification tags',
+           threatscore: 'threat score',
+           total_processes: 'total processes',
+           submitname: 'submit name',
+           environmentDescription: 'environment description',
+           isinteresting: 'interesting',
+           environmentId: 'environment id',
+           isurlanalysis: 'url analysis',
+           analysis_start_time: 'analysis start time',
+           total_signatures: 'total signatures'
+        };
+
+        var table = response.map(mapObject(tableMapper));
+
+        // create context from response
+        var context = {};
+
+        var contextMapper = {
+            sha1: 'SHA1',
+            sha256: 'SHA256',
+            md5: 'MD5',
+        };
+        context[outputPaths.file] = response.map(mapObject(contextMapper));
+        response.forEach(function(res) {
+           if(res.threatlevel && res.threatlevel > 1) {
+               addMalicious(context, outputPaths.file, {
+                    MD5: res.md5,
+                    SHA1: res.sha1,
+                    SHA256: res.sha256,
+                    Malicious: { Vendor: 'VXStream', Description: 'Score above ' + res.threatscore }
+                });
+           }
+        });
+        return createTableEntry('Scan Results:', response, table, context);
```

Comment:
might be better to try catch or anyway other to handle non-json response

Reply:
handled errors above it after `sendRequest` according to status code

Reply:
yes, in some cases the response is 200 but the body is not in json format (some proxies for example)

Reply:
In this specific integration?
At all integrations?

Reply:
at all. seen it with vectra and others